BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -44,6 +44,88 @@ public class NetworkStore: ObservableObject {
+          selectedChainId = network.chainId
+        }
+      }
+    }
+  }
+  
+  // MARK: - Custom Networks
+  
+  @Published var isAddingNewNetwork: Bool = false
+  
+  public func addCustomNetwork(_ network: BraveWallet.EthereumChain,
+                               completion: @escaping (_ accpted: Bool) -> Void) {
+    func addNetwors(_ network: BraveWallet.EthereumChain, completion: @escaping (_ accpted: Bool) -> Void) {
+      rpcService.add(network) { [self] chainId, status, message in
+        if status == .success {
+          // meaning removal is also succeeded if id exists. need to update the list by removing the old then adding the new
+          if let index = ethereumChains.firstIndex(where: { $0.id.lowercased() == network.id.lowercased() }) {
+            ethereumChains.remove(at: index)
+          }
+          ethereumChains.append(network)
+          isAddingNewNetwork = false
+          completion(true)
+        } else {
+          // meaning add custom network failed for some reason. We will not update `ethereumChains`
+          // Also add the the old network back on rpc service
+          if let oldNetwork = ethereumChains.first(where: { $0.id.lowercased() == network.id.lowercased() }) {
+            rpcService.add(oldNetwork) { _, _, _ in
+              isAddingNewNetwork = false
+              completion(false)
+            }
+          } else {
+            isAddingNewNetwork = false
+            completion(false)
+          }
+        }
+      }
+    }
+    
+    isAddingNewNetwork = true
+    if ethereumChains.contains(where: { $0.id.lowercased() == network.id.lowercased() }) {
+      removeNetworkForNewAddition(network) { [self] success in
+        guard success else {
+          isAddingNewNetwork = false
+          completion(false)
+          return
+        }
+        addNetwors(network, completion: completion)
+      }
+    } else {
+      addNetwors(network, completion: completion)
+    }
+  }
+  
+  /// This method will not update `ethereumChains`
+  private func removeNetworkForNewAddition(_ network: BraveWallet.EthereumChain,
+                                           completion: @escaping (_ success: Bool) -> Void) {
+    rpcService.removeEthereumChain(network.id) { success in
+      completion(success)
+    }
+  }
+  
+  public func removeCustomNetwork(_ network: BraveWallet.EthereumChain,
+                                  completion: @escaping (_ success: Bool) -> Void) {
+    rpcService.removeEthereumChain(network.id) { [self] success in
+      if success {
+        // check if its the current network, set mainnet the active net
+        if network.id.lowercased() == selectedChainId.lowercased() {
+          rpcService.setNetwork(BraveWallet.MainnetChainId, completion: { _ in })
+        }
+        // update `ethereumChains`
+        if let index = ethereumChains.firstIndex(where: { $0.id.lowercased() == network.id.lowercased() }) {
+          ethereumChains.remove(at: index)
+        }
```

Comment:
Same comment as above regarding manually removing instead of re-fetching all networks